### DISCUSSION TITLE
[API connection] No API available to connect

### DISCUSSION THREAD

**Discussion by Tungquantedge** [wazuh/wazuh#15962] [Q&A]
![image](https://user-images.githubusercontent.com/32952770/213109630-407be752-3739-4ce8-b418-6a5ce24b1219.png)
![image](https://user-images.githubusercontent.com/32952770/213111610-9cca89bf-9de6-4b86-accf-e905df10e1eb.png)
![image](https://user-images.githubusercontent.com/32952770/213111705-d16edcbd-3f66-47d4-b154-47547535d11f.png)
 
App started and no problems but in the next time errors showing

**Comment by Tungquantedge** [comment]
![image](https://user-images.githubusercontent.com/32952770/213112516-0bf8acbf-64db-470c-8850-4a5782b8bff4.png)
![image](https://user-images.githubusercontent.com/32952770/213112601-dea90a60-c8f4-4411-bc96-d38c5136d14c.png)

i saw reponse api return errors: 3099 - ERROR3099 - Wazuh not ready yet

**Comment by jctello** [comment]
Hi @Tungquantedge when restarting the manager the Wazuh API will wait until the Wazuh manager's modules are ready and the Wazuh Dashboard needs the Wazuh API to retrieve the status of the system as well as allowing you to modify the configuration from the web interface.

You may temporarily receive the error you're seeing under normal circumstances after a restart and retrying may resolve it.

If the problem persists there may be an underlying issue to address in the Wazuh manager. In order to troubleshoot it I recommend using the command line to first execute `systemctl restart wazuh-manager` and if the problem continues look for errors in the logs: `grep -iP 'error|critical' /var/ossec/logs/ossec.log`. 

A common issue may be having run out of disk space. If the problem persists you may also verify which version of Wazuh you're using, recent versions have addressed issues that may result in the error you see.

**Comment by simple-guy1** [comment]
@jctello I encounter the same error as @Tungquantedge, but only when I am trying to login into Wazuh through Cloudflare Connect which connects internet to my home on premise server. When I am connecting to Wazuh from my local network everything works. Could you please help me to identify where the issue could be? My setup looks like this:
- I am running wazuh in docker and `dashboard, manager and indexer` images are up an running
- After inserting login credentials I see the error `[API connection] No API available to connect`

When I check `grep -iP 'error|critical' /var/ossec/logs/ossec.log` in `wazuh-manager` docker image I get 
```
2023/08/25 05:23:50 wazuh-modulesd:vulnerability-detector: ERROR: (5513): CVE database could not be updated.
```

When I check developer console I see:
![Screenshot 2023-08-25 at 13 32 04](https://github.com/wazuh/wazuh/assets/124501083/2525e365-617b-4f14-ad79-0161d2364fab)

Where 
- `checked-stored-api` contains
```
{
    "statusCode": 200,
    "data": {
        "url": "https://wazuh.manager",
        "port": 55000,
        "username": "wazuh-wui",
        "password": "****",
        "run_as": false,
        "id": "1513629884013",
        "cluster_info": {
            "status": "disabled",
            "manager": "wazuh.manager",
            "cluster": "Disabled"
        },
        "secret": "****"
    },
    "idChanged": null
}
```
- `login` contains token
- `request` contains
```
{
    "statusCode": 401,
    "error": "Unauthorized",
    "message": "3013 - Request failed with status code 401"
}
```

Then in the end it ends with:
![Screenshot 2023-08-25 at 13 35 28](https://github.com/wazuh/wazuh/assets/124501083/34303d7b-27e9-45a2-95e4-06cebab9a537)
- `checked-stored-api` contains
```
{
    "statusCode": 429,
    "error": "Too Many Requests",
    "message": "3002 - Request failed with status code 429"
}
```

**Reply by simple-guy1** [reply]
And I was able to solve it 😄 LOL
The problem is solved through Edge Certificates setup where you select that you wanna always use HTTPS 
![Screenshot 2023-08-25 at 13 51 58](https://github.com/wazuh/wazuh/assets/124501083/2c00dce6-802c-4f52-b14f-44c0a062f247)
